### DISCUSSION TITLE
Is there a way to back fill lyrics? - Lidarr

### DISCUSSION THREAD

**Discussion by Deekerman** [RandomNinjaAtk/arr-scripts#199] [Q&A]
If you already had a large music collection before getting lidarr ext (just had lidarr previously) is there a way to get beets to go back and get the lyrics for everything that was there previously? I looked it up in beets and it seems there is a plugin that does that but it would be nice to have everything in one vs making another install of beets. 

Thoughts?

Thank you, 
D

**Comment by RandomNinjaAtk** [comment]
No, there isn't currently a method to do that. You can just use the installed version of beets and point it to your own config to do what you want.

You also could possibly write a script and add it to this repo as a way to do it in an automated fashion.

**Comment by Deekerman** [comment]
Thank you for the prompt response. Unfortunately, I am way to dumb to do any of that. I will either wait for it to be implemented  (if it is ever) or someone else to do it. I am barely able to run this. 

Thank you for what you have done, its been great.

**Reply by RandomNinjaAtk** [reply]
If your able to provide some links to documentation and etc, I can look into making something possibly.... No promises though.

**Reply by Deekerman** [reply]
This is what I had found, it seems there is a plugin that will look for lyrics. - https://beets.readthedocs.io/en/stable/plugins/lyrics.html

Then most specifically to the backfilling - https://beets.readthedocs.io/en/stable/plugins/lyrics.html#fetching-lyrics-manually

I have no expectations, the fact you are even willing to look at it is appreciated as my abilities are dismal.

**Reply by SickProdigy** [reply]
The thing with beets it is doesn't support .lrc as I know of. Only .epub, .pdf, and html. Although Maybe possible to go to html and just convert to .lrc. Was thinking that was needed for lyrics to show in navidrome. Right now I'm in the same boat have some songs synced with lyrics but not all. And kind of wanted the lyric file alongside the audio. Been trying to work something to gether and no avail yet. will post back if I find something.
Edit, this is kind of a config I came up with. You could just run this disable a bunch of the bs. Kind of works. Doesn't create the files though like I want yet. 
https://github.com/SickProdigy/beets-music-management/blob/main/config.yaml

**Reply by hockeygoalie35** [reply]
With beets 2.0, it now does support .lrc. See issue #204

**Comment by jasonpatrickellykrause** [comment]
I was looking for a way to do this today, specifically this function.

https://github.com/RandomNinjaAtk/arr-scripts/blob/main/lidarr/BeetsTagger.bash#L47

**Comment by jaycedk** [comment]
I use this for lyric's
https://github.com/tranxuanthang/lrcget

**Comment by hockeygoalie35** [comment]
Going forward, Lidarr extended will embed LRC lyrics, see #275 . lrcget is a great way to backfill what's been downloaded before.

**Reply by Deekerman** [reply]
Does that mean it will also backfill?

**Reply by hockeygoalie35** [reply]
No, that may be beyond the scope of the project. I can look into it though. Like I said, LRCGET is a great tool for now, it runs as a standard exe, so it’s very easy to use.

**Reply by Deekerman** [reply]
No problem, i dl LRCGET and it is scanning the library now. Thanks for the help.
